test/index.js

```diff
@@ -174,4 +174,56 @@ describe('preact-router', () => {
 				.and.calledWithExactly('/asdf');
 		});
 	});
+
+	describe('Async Route', () => {
+		class SampleTag extends Component {
+			render(){
+				return (<h1>hi</h1>);
+			}
+		}
+
+		it('should call the given function on mount', () => {
+			let getComponent = sinon.spy();
+			render(<AsyncRoute component={getComponent} />, document.createElement('div'));
+			expect(getComponent).called;
+		});
+
+		it('should render component when returned from a function', (done) => {
+			let containerTag = document.createElement('div');
+			let getComponent = function() {
+				return SampleTag;
+			};
+			render(<AsyncRoute component={getComponent} />, containerTag);
+
+			setTimeout(() => {
```

Comment:
@developit I wonder if there's a way to avoid such small timeouts... not a big fan
Without a small timeout it fails...